main/src/ca/uwaterloo/flix/language/ast/Type.scala

```diff
@@ -372,56 +372,56 @@ object Type {
     * A type variable.
     */
   @IntroducedBy(Kinder.getClass)
-  case class KindedVar(id: Int, kind: Kind, loc: SourceLocation, rigidity: Rigidity = Rigidity.Flexible, text: Option[String] = None) extends Type with Var with BaseType with Ordered[Type.KindedVar] {
+  case class KindedVar(sym: Symbol.TypeVarSym, kind: Kind, loc: SourceLocation, rigidity: Rigidity = Rigidity.Flexible, text: Option[String] = None) extends Type with Var with BaseType with Ordered[Type.KindedVar] {
```

Comment:
Looks good. I guess the really difficult question is how much of this data to leave here and how much to put into Symbol.TypeVar.